### PR TITLE
Fix linking errors with LIBSCSYNTH=ON

### LINKED ISSUE #5013: [scsynth] linking error with LIBSCSYNTH
## Environment

* SuperCollider version: develop
* Operating system: macOS (10.14)
* Other details (Qt version, audio driver, etc.): n/a

## Steps to reproduce

```
mkdir build
cd build
cmake .. -DLIBSCSYNTH=ON
cmake --build . --config Debug --target scsynth
```

## Expected vs. actual behavior

build should succeed, instead there's a linker error:

```
Ld build/server/scsynth/Debug/scsynth normal x86_64
    cd /Users/brianheim/git/sc2
    /Users/brianheim/git/sc2/build/launch-cxx -target x86_64-apple-macos10.14 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -L/Users/brianheim/git/sc2/build/server/scsynth/Debug -F/Users/brianheim/git/sc2/build/server/scsynth/Debug -filelist /Users/brianheim/git/sc2/build/server/scsynth/SuperCollider.build/Debug/scsynth.build/Objects-normal/x86_64/scsynth.LinkFileList -Xlinker -rpath -Xlinker /Users/brianheim/git/sc2/build/server/scsynth/Debug -Xlinker -object_path_lto -Xlinker /Users/brianheim/git/sc2/build/server/scsynth/SuperCollider.build/Debug/scsynth.build/Objects-normal/x86_64/scsynth_lto.o -Xlinker -no_deduplicate -stdlib=libc++ -Wl,-search_paths_first -Wl,-headerpad_max_install_names -stdlib=libc++ /Users/brianheim/git/sc2/build/server/scsynth/Debug/libscsynth.1.0.0.dylib /usr/lib/libpthread.dylib /Users/brianheim/git/sc2/build/external_libraries/Debug/libtlsf.a /usr/lib/libdl.dylib /usr/local/opt/libsndfile/lib/libsndfile.dylib -framework CoreAudio /Users/brianheim/git/sc2/build/external_libraries/Debug/libboost_system_lib.a /Users/brianheim/git/sc2/build/external_libraries/Debug/libboost_filesystem_lib.a -framework Accelerate -framework CoreServices -framework Foundation -framework AppKit -Xlinker -dependency_info -Xlinker /Users/brianheim/git/sc2/build/server/scsynth/SuperCollider.build/Debug/scsynth.build/Objects-normal/x86_64/scsynth_dependency_info.dat -o /Users/brianheim/git/sc2/build/server/scsynth/Debug/scsynth
Undefined symbols for architecture x86_64:
  "SC::Apple::EventLoop::run()", referenced from:
      EventLoop::run(std::__1::function<void ()>) in scsynth_main.o
  "SC::Apple::EventLoop::quit()", referenced from:
      EventLoop::run(std::__1::function<void ()>)::'lambda'()::operator()() const in scsynth_main.o
  "SC::Apple::EventLoop::setup()", referenced from:
      EventLoop::setup() in scsynth_main.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

EventLoop functions (and only EventLoop functions) should be included in scsynth rather than libscsynth.

### PR DESCRIPTION
**Pull request by brianlheim**
## Purpose and Motivation

This fixes #5013, as well as another linking error with LIBSCSYNTH

On linux, building with LIBSCSYNTH=ON and SYSTEM_BOOST=OFF fails with linking errors because
a static library is being linked into a shared library. solved by compiling the offending library with -fPIC.

On macOS, building with LIBSCSYNTH=ON fails with linking errors because SC_EventLoop.hpp functions used in
scsynth's main() are hidden, similar to what we encountered in #4992. This PR solves the issue the same way as
 #5012, by including the file in scsynth rather than libscsynth (and doing the equivalent change for supernova).
This required separating out the event loop-specific functions into a new header and implementation file since
disableAppNap is not used in quite the same way.

## Types of changes

- Bug fix

## To-do list

- [x] Code is tested - i've cherry picked these tentatively into #4906, where you can see that both LIBSCSYNTH builds
  succeed
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

**Comment by sonoro1234** [comment]
@brianlheim Sorry, I cant review this PR because my macos knowledge is really bad, I dont even have one.

**Comment by brianlheim** [comment]
@sonoro1234 -- ok, no worries. @Spacechild1 or @joshpar , perhaps you could give this a quick look?

File: common/SC_AppleEventLoop.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/************************************************************************
+ *
+ * Copyright 2013 Seth Nickell <snickell@gmail.com>
```
Comment (Spacechild1) [inline common/SC_AppleEventLoop.hpp:3]: Who is Seth Nickell? ;-)

File: common/SC_AppleEventLoop.mm [hunk 2]
```diff
@@ -0,0 +1,83 @@
+/************************************************************************
+ *
+ * Copyright 2013 Seth Nickell <snickell@gmail.com>
```
Comment (Spacechild1) [inline common/SC_AppleEventLoop.mm:3]: see above

**Comment by Spacechild1** [comment]
Apart from that it looks fine! The event loops certainly belongs to the scsynth/supernova apps. 

Note that people can still use the event loop with libscsynth, they just have to include `common/SC_AppleEventLoop.cpp` in their project and start the event loop manually (once).

File: common/SC_AppleEventLoop.hpp (hunk 1 above)
Comment (brianlheim) [reply common/SC_AppleEventLoop.hpp:3]: not sure! this was in the file before, and you kept it when you added your functions. do you want the copyright assignment then?

**Comment by joshpar** [comment]
I’m guessing event loop code came from another source. 
I’m ok adding attribution, but I prefer not to remove any previous ones.

/*
Josh Parmenter
www.realizedsound.net/josh
*/

**Comment by brianlheim** [comment]
> I’m guessing event loop code came from another source.

@Spacechild1 is the one who added the event loop code, that's why i asked if they wanted attribution. :) removing Seth Nickell's name here would not be removing attribution, because they did not contribute any of that code.

**Comment by joshpar** [comment]
I’m wondering if that is the author of some code (or even sample code) that was used for this bit of code that @Spacechild1 was including as reference.
If not, I’m totally fine removing it as well.

File: common/SC_AppleEventLoop.hpp (hunk 1 above)
Comment (Spacechild1) [reply common/SC_AppleEventLoop.hpp:3]: ah, I see! well, I would be flattered :-)

**Comment by Spacechild1** [comment]
> I’m wondering if that is the author of some code (or even sample code) that was used for this bit of code 

Seth Nickell was not involved in the event loop code. He probably wrote the `void disableAppNap();` which was the only function in the `SC_Apple.hpp` before I added the event loop.

**Comment by brianlheim** [comment]
@joshpar @Spacechild1 updated!

**Comment by bagong** [comment]
> Seth Nickell was not involved in the event loop code. He probably wrote the `void disableAppNap();` which was the only function in the `SC_Apple.hpp` before I added the event loop. The sample code I've referenced is by another author.

Yes, that's true, I remember Seth turning up out of nowhere when AppNap was giving people a little scare ;-)

**Pull request merged**